database/certificate.go

```diff
@@ -599,12 +599,17 @@ func (db *DB) GetValidationMapForCert(certID int64) (map[string]certificate.Vali
 	return certificate.GetValidityMap(ubuntu, mozilla, microsoft, apple, android), issuerId, nil
 }
 
-func (db *DB) GetCertPaths(cert *certificate.Certificate) (paths certificate.Paths, err error) {
+// GetCertPaths returns the various certificates paths from the current cert to roots.
+// It takes a certificate as argument that will be used as the start of the path, and
+// a genealogy slice that contains the list of known parents, to prevent looping. When
+// first invoked, the genealogy variable should be an empty []string.
```

Comment:
nit: I think the public API should not include the `[]string`, since after all we're asking to pass an empty slice the first call. I think we should instead have a private recursive function and make the public function create the slice for the caller.